### ISSUE TITLE
Wrong rows are selected for some tables

### ISSUE THREAD

**Issue by zroug** [sqlitebrowser/sqlitebrowser#2738]
Details for the issue
--------------------

#### What did you do?

I created a database with a single table using

```
CREATE TABLE data (
    key BLOB NOT NULL PRIMARY KEY,
    value BLOB NOT NULL
) WITHOUT ROWID"
```

I then inserted a row with key having type `BLOB` (not using the GUI).

#### What did you expect to see?

I should be able to delete or modify the row that I inserted using the GUI.

#### What did you see instead?

A a not existing (wrong) row is selected for deletion and other modifications. I think because DB4S uses `TEXT` instead of `BLOB` to select it.

Useful extra information
-------------------------

- A database that has this issue: [db.zip](https://github.com/sqlitebrowser/sqlitebrowser/files/6640906/db.zip)
- https://github.com/sqlitebrowser/sqlitebrowser/issues/2582 could be related but I think the fix doesn't apply to this particular issue.

#### What is your DB4S version?

- [x] 3.12.2

#### Did you also

- [x] Try out the latest nightly version: https://github.com/sqlitebrowser/sqlitebrowser#nightly-builds
- [x] Search for an existing similar issue: https://github.com/sqlitebrowser/sqlitebrowser/issues?utf8=%E2%9C%93&q=is%3Aissue%20

**Comment by mgrojo** [comment]
There is a contradiction in the description. I guess you actually inserted the row not using the GUI, otherwise it can be deleted. The problem, as you suggested, is that the type of the key is BLOB, but using the GUI we can only insert TEXT and manage TEXT. I'll take a look at it.

**Comment by zroug** [comment]
Yes, I meant that I inserted the row without using the GUI. Sorry if that was unclear, English is not my native language.

**Comment by mgrojo** [comment]
Ok.  I see now the misunderstanding. You wrote:
> I should be able to delete or modify the row that I inserted using the GUI.

I understood: "I should be able to delete or modify the row, which I inserted using the GUI."

What you meant in a less ambiguous way: "I should be able, using the GUI, to delete or modify the row that I inserted."

English isn't my native language either, which could contribute to misunderstanding too. Thanks for providing the database file, language of machines is sometimes clearer :smile:

**Comment by zroug** [comment]
Ah, I see, that is indeed ambiguous. I modified the issue according to your suggestion. Thanks for looking into the issue!

**Comment by mgrojo** [comment]
I've made some improvements, that you could try in the next nightly build, but take the comments in 6510bb104e3ec4c7419cc5558166920e75ec4046 into account, because if the value inserted as BLOB in the key can be taken as text data, DB Browser is confused and the operation still doesn't work. We have to figure out how to pass the actual value type to the code preparing the SQL statement.
